src/darker/tests/test_highlighting.py

```diff
@@ -1,50 +1,174 @@
+        "pygments color=False PY_COLORS=0 --no-color              ",
+        "pygments color=False PY_COLORS=0 --no-color tty          ",
+        "pygments color=False PY_COLORS=0    --color     USE_COLOR",
+        "pygments color=False PY_COLORS=0    --color tty USE_COLOR",
+        "pygments color=False PY_COLORS=1                USE_COLOR",
+        "pygments color=False PY_COLORS=1            tty USE_COLOR",
+        "pygments color=False PY_COLORS=1 --no-color              ",
+        "pygments color=False PY_COLORS=1 --no-color tty          ",
+        "pygments color=False PY_COLORS=1    --color     USE_COLOR",
+        "pygments color=False PY_COLORS=1    --color tty USE_COLOR",
+        "pygments color=True                             USE_COLOR",
+        "pygments color=True                         tty USE_COLOR",
+        "pygments color=True              --no-color              ",
+        "pygments color=True              --no-color tty          ",
+        "pygments color=True                 --color     USE_COLOR",
+        "pygments color=True                 --color tty USE_COLOR",
+        "pygments color=True  PY_COLORS=0                         ",
+        "pygments color=True  PY_COLORS=0            tty          ",
+        "pygments color=True  PY_COLORS=0 --no-color              ",
+        "pygments color=True  PY_COLORS=0 --no-color tty          ",
+        "pygments color=True  PY_COLORS=0    --color     USE_COLOR",
+        "pygments color=True  PY_COLORS=0    --color tty USE_COLOR",
+        "pygments color=True  PY_COLORS=1                USE_COLOR",
+        "pygments color=True  PY_COLORS=1            tty USE_COLOR",
+        "pygments color=True  PY_COLORS=1 --no-color              ",
+        "pygments color=True  PY_COLORS=1 --no-color tty          ",
+        "pygments color=True  PY_COLORS=1    --color     USE_COLOR",
+        "pygments color=True  PY_COLORS=1    --color tty USE_COLOR",
+    ],
+)
+def test_should_use_color(tmp_path: Path, params: str) -> None:
+    """Color output is used only if correct configuration options are in place"""
     modules = sys.modules.copy()
-    del modules["darker.highlighting"]
-    with patch.dict(sys.modules, modules, clear=True):
-        # pylint: disable=import-outside-toplevel
+    if "pygments " not in params:
+        del modules["darker.highlighting"]
+        # cause an ImportError for `import pygments`:
+        modules["pygments"] = None  # type: ignore[assignment]
+    with (tmp_path / "pyproject.toml").open("w") as pyproject:
+        print("[tool.darker]", file=pyproject)
+        if " color=True " in params:
+            print("color = true", file=pyproject)
+        if " color=False " in params:
+            print("color = false", file=pyproject)
+    env = {}
+    if " PY_COLORS=0 " in params:
+        env["PY_COLORS"] = "0"
+    if " PY_COLORS=1 " in params:
+        env["PY_COLORS"] = "1"
```

Comment:
This should be extend when allowing `true` and `false` for color.

Reply:
Ah, thanks @MatthijsBurgh, I didn't think of digging further to see whether other values than `0` and `1` are generally accepted for `PY_COLORS` by other Python tools. I'll double check – or do you have such examples readily in your mind?

Reply:
Based on a quick search on GitHub for [`PY_COLORS true`](https://github.com/search?p=2&q=PY_COLORS+true&type=Code) and [`PY_COLORS false`](https://github.com/search?p=2&q=PY_COLORS+false&type=Code), the only instance of non-`0`/`1` values is [here in Ansible `molecule`](https://github.com/ansible-community/molecule/blob/e78c820c6efdf09eb14432d15fca546730a1b37a/src/molecule/console.py#L45-L50):

It uses the [`to_bool()`](https://github.com/ansible-community/molecule/blob/e78c820c6efdf09eb14432d15fca546730a1b37a/src/molecule/console.py#L30-L38) helper, which has:
```python
    if isinstance(a, str):
        a = a.lower()
    if a in ("yes", "on", "1", "true", 1):
        return True
```

But since this is done not only when the `PY_COLORS` environment variable is encountered, but also `CLICOLOR`, `FORCE_COLOR` and `ANSIBLE_FORCE_COLOR`, it's not clear from which ones of these other values than `0` and `1` are to be expected.

Pytest documentation [says](https://docs.pytest.org/en/6.2.x/reference.html#envvar-PY_COLORS):
>**PY_COLORS**
>When set to `1`, pytest will use color in terminal output. When set to `0`, pytest will not use color. `PY_COLORS` takes precedence over `NO_COLOR` and `FORCE_COLOR`.

Textualize/rich#343 has some discussion on `PY_COLORS`, and only mentions `0`/`1`.

Many packages seem to rely on `py.io.TerminalWriter` to handle color output. Support for `PY_COLORS` seems to have been introduced in version 1.4.19 and is mentioned in the [change log](https://py.readthedocs.io/en/latest/changelog.html?highlight=PY_COLORS#changes-between-1-4-18-and-1-4-19):
>- check `PY_COLORS=1` or `PY_COLORS=0` to force coloring/not-coloring for `py.io.TerminalWriter()` independently from capabilities of the output file. Thanks Marc Abramowitz for the PR.

Based on the above evidence, I don't think we should jump into supporting other values than `PY_COLORS=0` and `PY_COLORS=1`. Would you agree?

Reply:
Haha, Good research. Then we should stick to `0/1`.

Reply:
pypa/pip#10909 is so far the "highest profile" discussion I've found on this topic.

It seems that [pip](https://github.com/pypa/pip) and [colorama](https://github.com/tartley/colorama) are going to support only `FORCE_COLOR=<anything>` and `NO_COLOR=<anything>`, but not `PY_COLORS`.

On the other hand, Pytest does support `PY_COLORS={1|0}` and `NO_COLOR=<anything>` (with `PY_COLORS` taking precedence).

So in the end it may be best to do the same as Pytest:
- `PY_COLORS={1|0}` (taking precedence over other options)
- `NO_COLOR=<anything>`
- and possibly `FORCE_COLOR=<anything>` (similar to [pip](https://github.com/pypa/pip) and [colorama](https://github.com/tartley/colorama))

As for the `true`/`false` discussion, I'm going to make the call to only support `PY_COLORS={0|1}` for now. Thanks @MatthijsBurgh for initiating the discussion and motivating me to look further into the topic!

Reply:
I added #355 for adding `NO_COLOR` (and possibly `FORCE_COLOR`) support in the future.